pkg/cloudprovider/aws/zz_generated.pricing.go

```diff
@@ -0,0 +1,546 @@
+	"r6gd.xlarge": 0.230400,
+	// r6i family
+	"r6i.12xlarge": 3.024000, "r6i.16xlarge": 4.032000, "r6i.24xlarge": 6.048000, "r6i.2xlarge": 0.504000,
+	"r6i.32xlarge": 8.064000, "r6i.4xlarge": 1.008000, "r6i.8xlarge": 2.016000, "r6i.large": 0.126000,
+	"r6i.metal": 8.064000, "r6i.xlarge": 0.252000,
+	// r6id family
+	"r6id.12xlarge": 3.628800, "r6id.16xlarge": 4.838400, "r6id.24xlarge": 7.257600, "r6id.2xlarge": 0.604800,
+	"r6id.32xlarge": 9.676800, "r6id.4xlarge": 1.209600, "r6id.8xlarge": 2.419200, "r6id.large": 0.151200,
+	"r6id.metal": 9.676800, "r6id.xlarge": 0.302400,
+	// t1 family
+	"t1.micro": 0.020000,
+	// t2 family
+	"t2.2xlarge": 0.371200, "t2.large": 0.092800, "t2.medium": 0.046400, "t2.micro": 0.011600,
+	"t2.nano": 0.005800, "t2.small": 0.023000, "t2.xlarge": 0.185600,
+	// t3 family
+	"t3.2xlarge": 0.332800, "t3.large": 0.083200, "t3.medium": 0.041600, "t3.micro": 0.010400,
+	"t3.nano": 0.005200, "t3.small": 0.020800, "t3.xlarge": 0.166400,
+	// t3a family
+	"t3a.2xlarge": 0.300800, "t3a.large": 0.075200, "t3a.medium": 0.037600, "t3a.micro": 0.009400,
+	"t3a.nano": 0.004700, "t3a.small": 0.018800, "t3a.xlarge": 0.150400,
+	// t4g family
+	"t4g.2xlarge": 0.268800, "t4g.large": 0.067200, "t4g.medium": 0.033600, "t4g.micro": 0.008400,
+	"t4g.nano": 0.004200, "t4g.small": 0.016800, "t4g.xlarge": 0.134400,
+	// u-12tb1 family
+	"u-12tb1.112xlarge": 109.200000,
+	// u-3tb1 family
+	"u-3tb1.56xlarge": 27.300000,
+	// u-6tb1 family
+	"u-6tb1.112xlarge": 54.600000, "u-6tb1.56xlarge": 46.403910,
+	// u-9tb1 family
+	"u-9tb1.112xlarge": 81.900000,
+	// vt1 family
+	"vt1.24xlarge": 5.200000, "vt1.3xlarge": 0.650000, "vt1.6xlarge": 1.300000,
+	// x1 family
+	"x1.16xlarge": 6.669000, "x1.32xlarge": 13.338000,
+	// x1e family
+	"x1e.16xlarge": 13.344000, "x1e.2xlarge": 1.668000, "x1e.32xlarge": 26.688000, "x1e.4xlarge": 3.336000,
+	"x1e.8xlarge": 6.672000, "x1e.xlarge": 0.834000,
+	// x2gd family
+	"x2gd.12xlarge": 4.008000, "x2gd.16xlarge": 5.344000, "x2gd.2xlarge": 0.668000, "x2gd.4xlarge": 1.336000,
+	"x2gd.8xlarge": 2.672000, "x2gd.large": 0.167000, "x2gd.medium": 0.083500, "x2gd.metal": 5.878400,
+	"x2gd.xlarge": 0.334000,
+	// x2idn family
+	"x2idn.16xlarge": 6.669000, "x2idn.24xlarge": 10.003500, "x2idn.32xlarge": 13.338000,
+	"x2idn.metal": 13.338000,
+	// x2iedn family
+	"x2iedn.16xlarge": 13.338000, "x2iedn.24xlarge": 20.007000, "x2iedn.2xlarge": 1.667250,
+	"x2iedn.32xlarge": 26.676000, "x2iedn.4xlarge": 3.334500, "x2iedn.8xlarge": 6.669000,
+	"x2iedn.metal": 26.676000, "x2iedn.xlarge": 0.833630,
+	// x2iezn family
+	"x2iezn.12xlarge": 10.008000, "x2iezn.2xlarge": 1.668000, "x2iezn.4xlarge": 3.336000,
+	"x2iezn.6xlarge": 5.004000, "x2iezn.8xlarge": 6.672000, "x2iezn.metal": 10.008000,
+	// z1d family
+	"z1d.12xlarge": 4.464000, "z1d.2xlarge": 0.744000, "z1d.3xlarge": 1.116000, "z1d.6xlarge": 2.232000,
+	"z1d.large": 0.186000, "z1d.metal": 4.464000, "z1d.xlarge": 0.372000,
+}
+
+var initialSpotPrices = map[string]float64{
```

Comment:
I don't think there's much of a purpose to store static spot pricing data since, unlike OD, it won't be the same relative ranking within different regions. I think it might be better to just assume OD pricing if not able to pull spot pricing.

Reply:
Ah, that makes sense, I didn't think to fall back to OD pricing if we can't get spot. We'd have to reconsider once we support pricing per offering,  maybe spot = od*some discount at that point?

Reply:
If we assume OD pricing, we'll still be able to leverage fleet for price selection on provisioning, we just lose the ability to defrag spot until pricing comes back, which seems acceptable. We can also do something like label the instance with its price when its created to avoid bad defrags.